src/config_file.c

```diff
@@ -326,6 +326,9 @@ static int config_refresh(git_config_backend *cfg)
 	int error, modified;
 	uint32_t i;
 
+	if (b->header.parent.readonly)
+		return 0;
```

Comment:
Hm. Not all callers do check that, most notably `config_set` and `config_delete` functions don't. So this does change behavior, no? This change definitly looks a bit scary to me

Reply:
`config_set` and `config_delete` only are set as the set and delete functions for the "live"/read-write object so. The snapshots (i.e. when `readonly` would be true) sets its own functions which return with an error.

So `config_set` and `config_delete` would only be called when `readonly` is false and thus simply continue.

Reply:
Oh, missed that, thanks for pointing out. The only thought left is that it feels a bit weird to return success even if we bail out early. I'd say returning an error code is a bit cleaner, as that may not leave us wondering at a later point why just some random configs are never being refreshed. As callers don't call `refresh` anyway when the config is read-only, this shouldn't have any negative impact right now but feels safer for the future.

Reply:
We do have a function that we use in case you accidentally end up calling set and friends on a snapshot. I'll make it return that.